fdbserver/ClusterController.actor.cpp

```diff
@@ -477,14 +505,45 @@ class ClusterControllerData {
 			const auto& worker_details = worker_info.details;
 			auto fitness = worker_details.processClass.machineClassFitness(ProcessClass::TLog);
 
-			if (std::find(exclusionWorkerIds.begin(), exclusionWorkerIds.end(), worker_details.interf.id()) !=
-			        exclusionWorkerIds.end() ||
-			    !workerAvailable(worker_info, checkStable) ||
-			    conf.isExcludedServer(worker_details.interf.addresses()) || fitness == ProcessClass::NeverAssign ||
-			    (!dcIds.empty() && dcIds.count(worker_details.interf.locality.dcId()) == 0) ||
-			    (!allowDegraded && worker_details.degraded)) {
+			if (fitness == ProcessClass::NeverAssign) {
+				logWorkerUnavailable(
+				    SevDebug, id, "complex", "Worker's fitness is NeverAssign", worker_details, fitness, dcIds);
 				continue;
 			}
+			// Logging every possible options is a lot for every recruitment; logging all of the options with GoodFit or
+			// BestFit may work because there should only be like 30 tlog class processes.
+			if (fitness == ProcessClass::BestFit || fitness == ProcessClass::BestFit) {
```

Comment:
One of them should be GoodFit.

Put this condition inside `logWorkerUnavailable` above to guard printing the log event. Otherwise, this is going to `include` all workers that don't have BestFit and GoodFit (see line 548 below), and change the tlog recruitment logic.

Reply:
Good point. I don't know what I was thinking.